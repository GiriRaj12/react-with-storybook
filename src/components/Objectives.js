import React, { useState, useEffect } from 'react';

function Objectives() {
    const [objectives, setObjectives] = useState({});

    useEffect(() => {
        getIndividualObjectives(setObjectives);
    }, []
    );

    return <div className="individual-objectives">
        {designIndividualObjectives("Objectives", "Total number of Objectives", objectives.totalObjectives, "count count-blue")}
        {designIndividualObjectives("Objectives achieved", "Objectives over 70% progress", objectives.totalObjectivesAchieved, "count count-green")}
        {designIndividualObjectives("Objectives at risk", "Objectives below 20% progress", objectives.totalObjectivesAtRisk, "count count-red")}
        {designIndividualObjectives("Objectives on track", "Objectives with 20% - 70% progress", objectives.totalObjectivesOnTrack, "count count-orange")}
    </div>
}

function getIndividualObjectives(callback) {
    fetch("http://localhost:8080/objectives")
        .then(res => res.json())
        .then(res => {
            callback(res);
        })
}

function designIndividualObjectives(title, content, achieved, countClass) {
    return (<div className="individual-objective-containers">
        <div className="container-titles">
            {title}
        </div>
        <div className="objective-content">
            {content}
        </div>
        <div className={countClass}>
            {achieved}
        </div>
    </div>)
}
export default Objectives;