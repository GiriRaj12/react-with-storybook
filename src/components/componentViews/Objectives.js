import React, { useState, useEffect } from 'react';
import {get} from '../services/HttpSrevices.js';
import Log from '../services/Log.js';

function Objectives() {
    const [objectives, setObjectives] = useState({});

    useEffect(() => {
        getIndividualObjectives(setObjectives);
    }, []
    );

    return <div className="individual-objectives">
        {designIndividualObjectives("Objectives", "Total number of Objectives", objectives.totalObjectives, "count count-blue")}
        {designIndividualObjectives("Objectives achieved", "Objectives over 70% progress", objectives.totalObjectivesAchieved, "count count-green")}
        {designIndividualObjectives("Objectives at risk", "Objectives below 20% progress", objectives.totalObjectivesAtRisk, "count count-red")}
        {designIndividualObjectives("Objectives on track", "Objectives with 20% - 70% progress", objectives.totalObjectivesOnTrack, "count count-orange")}
    </div>
}

function getIndividualObjectives(callback) {
    get('/objectives').then(res => {
        Log.warn(res,'Objectives component');
        callback(res);
    });
}

function designIndividualObjectives(title, content, achieved, countClass) {
    return (<div className="individual-objective-containers">
        <div className="container-titles">
            {title}
        </div>
        <div className="objective-content">
            {content}
        </div>
        <div className={countClass}>
            {achieved}
        </div>
    </div>)
}
export default Objectives;