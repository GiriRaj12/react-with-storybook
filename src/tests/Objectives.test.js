import React from 'react';
import { render } from '@testing-library/react';
import Objectives from '../components/Objectives.js';
import renderer from 'react-test-renderer';


test('Objective Element test', () => {
    const component = renderer.create(
        <Objectives></Objectives>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Objectives Element to be in the document', () => {
    const { container } = render(<Objectives></Objectives>);
    expect(container.firstChild.childNodes.length).toBe(4);
})