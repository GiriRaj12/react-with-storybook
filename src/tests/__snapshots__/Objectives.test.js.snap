// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objective Element test 1`] = `
<div
  className="individual-objectives"
>
  <div
    className="individual-objective-containers"
  >
    <div
      className="container-titles"
    >
      Objectives
    </div>
    <div
      className="objective-content"
    >
      Total number of Objectives
    </div>
    <div
      className="count count-blue"
    />
  </div>
  <div
    className="individual-objective-containers"
  >
    <div
      className="container-titles"
    >
      Objectives achieved
    </div>
    <div
      className="objective-content"
    >
      Objectives over 70% progress
    </div>
    <div
      className="count count-green"
    />
  </div>
  <div
    className="individual-objective-containers"
  >
    <div
      className="container-titles"
    >
      Objectives at risk
    </div>
    <div
      className="objective-content"
    >
      Objectives below 20% progress
    </div>
    <div
      className="count count-red"
    />
  </div>
  <div
    className="individual-objective-containers"
  >
    <div
      className="container-titles"
    >
      Objectives on track
    </div>
    <div
      className="objective-content"
    >
      Objectives with 20% - 70% progress
    </div>
    <div
      className="count count-orange"
    />
  </div>
</div>
`;
